### ISSUE TITLE
Erroneous errors when creating generic class instance

### ISSUE THREAD

**Issue by bpatram** [babel/babel-eslint#660]
Hello, I am using eslint 5.1.0 (current latest) with the babel-eslint parser along with eslint-plugin-flowtype 2.50.0 (current latest) using the recommended configuration (`plugin:flowtype/recommended`) and the airbnb preset (`eslint-config-airbnb`)

The code snippet below returns errors which I believe are not correct:

```javascript
// @flow

class MyClass<T> {
  foo: T;

  constructor(foo: T) {
    this.foo = foo;
  }
}

export default new MyClass<string>('test');
```

Running this through the linter returns the following errors:
```
  11:16  error  Missing '()' invoking a constructor  new-parens
  11:27  error  Unexpected mix of '<' and '>'        no-mixed-operators
  11:27  error  Infix operators must be spaced       space-infix-ops
  11:28  error  'string' is not defined              no-undef
  11:34  error  Infix operators must be spaced       space-infix-ops
  11:34  error  Unexpected mix of '<' and '>'        no-mixed-operators
```

It seems to be parsing the type def brackets `<` `>` as infix operators GT and LT which causes some issues (such as thinking `string` is a variable and that the constructor never gets invoked).

I have pulled this issue out into its own repo here: https://github.com/bpatram/eslint-flow-generics which can be pulled down and replicated.

The above example throws no errors in the [flow type example I made here](https://flow.org/try/#0PTAEAEDMBsHsHcBQiDG0CGBnToCyBPAYQ2wB4AVAPlAG9FRRJZYAuUcgbmQZVgDtMAFwBOAVxSDYwgBRNW7AJS16DUIIAWAS0wA6OaAC8jZlwYBfRBcQBTAB4AHKYNAATa5HSjozvtfh4iEkxSIWFNPgBzSmkAckFrIRiFDiA).

**Comment by bpatram** [comment]
Looking into this further. It seems to be a babel parse/compilation issue. When compiled via babel it outputs: 
```javascript
class MyClass {

  constructor(foo) {
    this.foo = foo;
  }
}

export default new MyClass() < string > 'test';
```

Which as you can see, the `<string>` type is not being correctly striped out

**Comment by bpatram** [comment]
Closing as this is a babel related issue, not babel-eslint. See upstream issue: https://github.com/babel/babel/issues/8332
